### PR TITLE
Better support for make TIMED=1 on Windows

### PR DESCRIPTION
**Pull request by JasonGross**
This fixes [bug #5619](https://coq.inria.fr/bugs/show_bug.cgi?id=5619)

### CODE REVIEW

**Review by gares — APPROVED** [review]
I don't see why printing real time (that is the most important one IMHO) has something to do with windows, but I'm all for it.

Thanks!

**Comment by JasonGross** [comment]
I think user time is more stable across runs, and is robust against pausing the process, than real time (though of course user time doesn't include things like how long it takes to load the `Requires`).  What it has to do with Windows is that cygwin's `/usr/bin/time` always reports 0 for user time.

Related question: once this gets merged, should I have the timing scripts always use real time, rather than having them use it only when user time is 0?

**Comment by JasonGross** [comment]
Rebased on top of newly fixed trunk

**Comment by maximedenes** [comment]
@gares why do you say real time is better than user time? I thought it was the other way around, at least for reliable benchmarks.

**Comment by ejgallego** [comment]
I am not expert on bechmarking but both of them have drawback in Linux I believe (you used to be a ble to make a program that consumes little user time but it is very slow)

**Comment by JasonGross** [comment]
> you used to be a ble to make a program that consumes little user time but it is very slow

This is very easy; open a gigantic file and mmap its contents, and I think all the time will be spent in sys.  Also, I suspect `sleep 2` might have very little time spent in user-space.

**Comment by gares** [comment]
User time is the theory, clock time is the practice :-P

More seriously, I've no idea why, but user time has never been accurate. Wall clock is, but you may get some interference when the system is loaded by other tasks.

As jason points out, if I add a sleep 10 in a busy loop, I'll hardly notice it by looking at user time only. Ditto for I/O or loos/gain in data locality and CPU cache-hit/miss.

To sum up, both are relevant, even if they are almost always very close.

**Comment by maximedenes** [comment]
I am right that this PR replaces the printing of the user time by real time? Is there a reason to not print both?

**Comment by JasonGross** [comment]
No, this PR replaces user+mem with real+user+sys+mem.  This PR makes it so that there is strictly more information.

**Comment by maximedenes** [comment]
Sorry, I misread, indeed!
